keys/keys.go

```diff
@@ -289,5 +289,10 @@ func DecodeRangeMetaKey(key proto.Key) (proto.Key, proto.Key) {
 	}
 	// Otherwise find the first entry greater than the given key in the same meta prefix.
 	metaPrefix := proto.Key(key[:len(Meta1Prefix)])
+	metaPrefixMax := proto.MakeKey(metaPrefix, proto.KeyMax)
+
+	if key.Equal(metaPrefixMax) {
+		return key, metaPrefix.PrefixEnd()
+	}
```

Comment:
`TestInternalRangeLookup` is doing an RPC and a whole bunch of other stuff. I'm suggesting adding `TestDecodeRangeMetaKey` to `keys_test.go`.